pkg/config/validation/validation.go

```diff
@@ -1025,10 +1030,12 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 			bind := i.GetBind()
 			errs = appendValidation(errs, validateSidecarIngressPortAndBind(i.Port, bind))
 
-			if _, found := portMap[i.Port.Number]; found {
+			p := ipPortPair{ip: bind, port: i.Port.Number}
```

Comment:
I'm realizing that having a `bind` of `0.0.0.0` and `W.X.Y.X` could actually cause a conflict especially if SO_REUSEADDR/SO_REUSEPORT is being set.

perhaps its just safer to set this on the egress part of the sidecar unless you think its worth trying to support this on ingress side too.

Reply:
~~perhaps its just safer to set this on the egress part of the sidecar unless you think its worth trying to support this on ingress side too.~~

actually in both ingress/egress, if there are two listeners one with `0.0.0.0` and the other with `W.X.Y.X` with the same ports, this should not be allowed.

Reply:
@dmichel1 could you please elaborate the problem with SO_REUSEADDR/SO_REUSEPORT?

How about having multiple Sidecar's egress listeners  with empty `bind` but same port number for TLS/HTTPS protocol? they suppose to be multiple filter chain in same listener, but each filter chain can have separate TLS context. 

IMO it's reasonable to allow sidecar egress listener using same port number for TLS/HTTPS, because they can be multiplexed by FilterChainMatch with different SNI. If sidecar's outbound listener builder doesn't generate  listener config for doing that, it can be changed accordingly.

Looks like https://github.com/istio/istio/issues/38370 is somehow related to this.